### PR TITLE
refactor: changed way internal logic works and the exposed api

### PR DESCRIPTION
**Pull request by DaniTulp**
#1243

### CODE REVIEW

File: packages/@tinacms/react-sidebar/src/sidebar.ts [hunk 1]
```diff
@@ -17,13 +17,14 @@ limitations under the License.
...
 import { EventBus, Callback } from '@tinacms/core'
+import { NoFormsPlaceholder } from './components/NoFormsPlaceHolder'
 import * as React from 'react'
 
 export interface SidebarStateOptions {
   hidden?: boolean
   position?: SidebarPosition
   buttons?: SidebarButtons
-  emptyState?: React.ElementType
+  placeholder?: React.ReactNode
```
Comment (ncphillips) [inline packages/@tinacms/react-sidebar/src/sidebar.ts:27]: I think I would rather pass the component than a node. Something like:

```suggestion
  placeholder?: React.FC
```

Comment (ncphillips) [reply packages/@tinacms/react-sidebar/src/sidebar.ts:27]: Besides that this great! 🙌

Comment (DaniTulp) [reply packages/@tinacms/react-sidebar/src/sidebar.ts]: Yep sounds good!

**Comment by ncphillips** [comment]
@all-contributors please add @DaniTulp for docs

**Pull request merged**